lib/factor/factorQuadratic.js

```diff
@@ -0,0 +1,127 @@
+        constantTermRootValue % 1 === 0) {
+      const polyTerm = Node.Creator.polynomialTerm(symbol, null, firstTermCoeffRootNode);
+      const firstParen = Node.Creator.parenthesis(
+        Node.Creator.operator('+', [polyTerm, constantTermRootNode]));
+      const secondParen = Node.Creator.parenthesis(
+        Node.Creator.operator('-', [polyTerm, constantTermRootNode]));
+
+      // create node in difference of squares form
+      const newNode = Node.Creator.operator('*', [firstParen, secondParen], true);
+      return Node.Status.nodeChanged(
+        ChangeTypes.FACTOR_DIFFERENCE_OF_SQUARES, node, newNode);
+    }
+  }
+  else {
+    // check if perfect square: are a and c squares and b = 2*sqrt(a)*sqrt(c)
+
+    // if the second term is negative, then the constant in the parens is subtracted
+    // i.e. x^2 -2x + 1 -> (x-1)^2
+    const secondTermCoeffValue = secondTermNode.getCoeffValue();
+    if (secondTermCoeffValue < 0) {
+      constantTermRootValue = constantTermRootValue * -1;
+      constantTermRootNode = Negative.negate(constantTermRootNode);
+    }
+
+    const perfectProduct = 2 * firstTermCoeffRootValue * constantTermRootValue;
+    if (firstTermCoeffRootValue % 1 === 0 &&
+        constantTermRootValue % 1 === 0 &&
+        secondTermCoeffValue === perfectProduct) {
+      const polyTerm = Node.Creator.polynomialTerm(symbol, null, firstTermCoeffRootNode);
+      const paren = Node.Creator.parenthesis(
+        Node.Creator.operator('+', [polyTerm, constantTermRootNode]));
+      const exponent = Node.Creator.constant(2);
+
+      // create node in perfect square form
+      const newNode = Node.Creator.operator('^', [paren, exponent]);
+      return Node.Status.nodeChanged(
+        ChangeTypes.FACTOR_PERFECT_SQUARE, node, newNode);
+    }
+    else if (firstTermCoeffValue === 1) {
+      // try sum/product rule: find a factor pair of c that adds up to b
+      const factorPairs = ConstantFactors.getFactorPairs(constantTermValue, true);
+      for (const pair of factorPairs) {
+        if (pair[0] + pair[1] === secondTermCoeffValue) {
+          const polyTerm = Node.Creator.polynomialTerm(symbol, null, firstTermCoeffRootNode);
+          const firstParen = Node.Creator.parenthesis(
+            Node.Creator.operator('+', [polyTerm, Node.Creator.constant(pair[0])]));
+          const secondParen = Node.Creator.parenthesis(
+            Node.Creator.operator('+', [polyTerm, Node.Creator.constant(pair[1])]));
+
+          // create a node in the general factored form for expression
+          const newNode = Node.Creator.operator('*', [firstParen, secondParen], true);
+          return Node.Status.nodeChanged(
+            ChangeTypes.FACTOR_SUM_PRODUCT_RULE, node, newNode);
+        }
+      }
+    }
+  }
+
+  // TODO: quadratic formula
+  // a(x - (-b + sqrt(b^2 - 4ac)) / 2a)(x - (-b - sqrt(b^2 - 4ac)) / 2a)
+
+  return Node.Status.noChange(node);
+}
+
+module.exports = factorQuadratic;
```

Comment:
this functionality won't exposed in mathsteps yet, right?

should we add it to the top level index.js, or wait until we figure out what a more general factor interface will look like?